src/coreComponents/physicsSolvers/fluidFlow/IsothermalCompositionalMultiphaseBaseKernels.hpp

```diff
@@ -720,6 +743,9 @@ class ElementBasedAssemblyKernel
         real64 const dPhaseCompAmount_dP = dPhaseAmount_dP * phaseCompFrac[ip][ic]
                                            + phaseAmount * dPhaseCompFrac[ip][ic][Deriv::dP];
 
+        GEOS_ASSERT_MSG( ic == 0 && LvArray::math::abs( dPhaseCompAmount_dP ) < minDensForDivision,
+                         "Zero diagonal in Jacobian" );
```

Comment:
@klevzoff is it allowed or bad?

Reply:
Having an assertion in kernel is fine, keep in mind it is only enabled in debug builds. You want to negate the condition here though, `GEOS_ASSERT` works opposite of `GEOS_ERROR_IF`.

Reply:
Just corrected, thanks!